lib/elixir/lib/enum.ex

```diff
@@ -3856,8 +3861,13 @@ defimpl Enumerable, for: Map do
   end
 
   def slice(map) do
-    size = map_size(map)
-    {:ok, size, &Enumerable.List.slice(:maps.to_list(map), &1, &2, size)}
+    case map_size(map) do
```

Comment:
Sorry but I don't see how this is an improvement because the caller will always match on the size and never call the function, so the contents of the function does not matter. :)

Reply:
it is saving a call to `:maps.to_list(map)` when we know map is empty.

Reply:
But if the size is zero we never call the slice function. Do we?

Reply:
also, when calling slice/1 it is saving a call to Enumerable.List.slice/4.